framework/test/bundles/libH/TestBundleH.cpp

```diff
@@ -66,19 +67,20 @@ class TestProduct2 : public TestProduct, public TestBundleH2
 class TestBundleHPrototypeServiceFactory : public PrototypeServiceFactory
```

Comment:
The service factory implementation could call back into the framework (by e.g. getting another service). This would test if you get a deadlock, which should happen with the proposed changes.